cpp/thirdparty/versions.txt

```diff
@@ -53,6 +53,8 @@ ARROW_AWS_LC_BUILD_VERSION=v1.3.0
 ARROW_AWS_LC_BUILD_SHA256_CHECKSUM=ae96a3567161552744fc0cae8b4d68ed88b1ec0f3d3c98700070115356da5a37
 ARROW_AWSSDK_BUILD_VERSION=1.10.55
 ARROW_AWSSDK_BUILD_SHA256_CHECKSUM=2d552fb1a84bef4a9b65e34aa7031851ed2aef5319e02cc6e4cb735c48aa30de
+ARROW_AZURE_SDK_BUILD_VERSION=azure-identity_1.5.1
```

Comment:
Despite the confusing version name this is still the whole azure SDK including core, keyvault, storage-common, etc.

Reply:
Could you add the information as a comment for the line for further changes?
BTW, why did you chooze `azure-identity_*`? It seems that azure-storage-files-datalake is the important package for us.

Reply:
I chose [azure-identity_1.5.1](https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-identity_1.5.1) because it was the first release with functioning managed identity support. When I discovered that managed identity was broken it was also the latest release at that time. 

To give an example [azure-core_1.10.1](https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-core_1.10.1) and [azure-identity_1.5.1](https://github.com/Azure/azure-sdk-for-cpp/releases/tag/azure-identity_1.5.1) are consecutive releases. It seems to me that it would be better to think about them as `1.0.0` and `1.0.1`. 

I've added the comment to the code.